pinot-spi/src/main/java/org/apache/pinot/spi/accounting/ThreadResourceUsageProvider.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.accounting;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The {@code ThreadResourceUsageProvider} class providing the functionality of measuring the CPU time
+ * and allocate bytes for the current thread.
+ */
+public class ThreadResourceUsageProvider {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ThreadResourceUsageProvider.class);
+  private static final String SUN_THREAD_MXBEAN_CLASS_NAME = "com.sun.management.ThreadMXBean";
+  private static final String SUN_THREAD_MXBEAN_IS_THREAD_ALLOCATED_MEMORY_SUPPORTED_NAME
+      = "isThreadAllocatedMemorySupported";
+  private static final String SUN_THREAD_MXBEAN_IS_THREAD_ALLOCATED_MEMORY_ENABLED_NAME
+      = "isThreadAllocatedMemoryEnabled";
+  private static final String SUN_THREAD_MXBEAN_GET_BYTES_ALLOCATED_NAME = "getThreadAllocatedBytes";
+  private static final Method SUN_THREAD_MXBEAN_GET_BYTES_ALLOCATED_METHOD;
+
+  private static final ThreadMXBean MX_BEAN = ManagementFactory.getThreadMXBean();
+  private static final boolean IS_CURRENT_THREAD_CPU_TIME_SUPPORTED = MX_BEAN.isCurrentThreadCpuTimeSupported();
+  private static final boolean IS_THREAD_ALLOCATED_MEMORY_SUPPORTED;
+  private static final boolean IS_THREAD_ALLOCATED_MEMORY_ENABLED;
+  private static boolean _isThreadCpuTimeMeasurementEnabled = false;
+  private static boolean _isThreadMemoryMeasurementEnabled = false;
+  private final long _startTimeNs;
```

Comment:
(nit) suggest adding an empty line before 49 for readability

Reply:
Done